### DISCUSSION TITLE
Thermal Mode (Thermal Detector Mode)

### DISCUSSION THREAD

**Discussion by F1AWings** [iNavFlight/inav#10559] [Ideas]
Any way of having a thermal detector setup on a small and light electric glider with the small electric motor to be used only when necessary? 
Explanation: The model will start to loiter at 75meters over the RTH location, the motor will be turned off at this altitude, it will circle with a 50mt radius as is set up in the RTH setup. If the glider goes down to 40mt, the motor will be turned on and the FC will take the glider back to 75mt and turn motor off again, this will be repeated until any of the TX sticks are moved. (Altitudes and loiter radius are all to be programmable) 
If the glider is gaining altitude due to thermals (and when the motor is off) the FC will turn the LED flasher connected to the FC. If the model is sinking when the motor is of the FC will turn the LED flasher off.
Is there anyway to do this with the current releases of Inav or the upcoming version 8?
This would be a great way of getting to know a flight location for thermal activity exclusively for non-contest use when getting ready for all sorts of thermal flying contests as the use of such a device is usually banned during thermal based contests.

**Comment by MrD-RC** [comment]
It doesn't quite work that way as thermals can move. So loitering around a single point would only work for a certain period of time. 

I am working on a total energy vario. Which will make finding thermals easier. That will hopefully be in INAV 9.0. I'm currently building an airplane to test with in the new year.

**Reply by F1AWings** [reply]
Great. Keep up the good work.
I have been flying competition free flight models (F1A) along with RC gliders and other RC models for the past 30 years and the altitudes and radiuses I provided are purely for explanation of the concept. If the radius of the flight can be selected with the option to turn off the motor when a certain altitude is reached and turned back on after a certain minimum altitude is reached while keeping the model in a certain geozone and having a feedback (for thermal activity) back to the people on the ground with an LED that is the whole idea.

**Comment by sensei-hacker** [comment]
You have mostly done it already by describing it in step-by-step detail. The only remaining step for you to be able to use it would be to transfer what you wrote above to the Programming tab.

In the Programming tab, you would probably set a counter that increments every loop. When the counter reaches 10, check whether the altitude has increased or decreased and do the actions you wrote. Then reset the counter to 0.

**Reply by F1AWings** [reply]
Thanks. I thought it could be possible under the programming tab but I could not come up with the necessary steps, can someone please help?

**Reply by sensei-hacker** [reply]
> Thanks. I thought it could be possible under the programming tab but I could not come up with the necessary steps, can someone please help?


Line 10:  Check if the current altitude is greater than global variable 0.

Line 11: Set global variable 0 to the current altitude

Line 15-xx do stuff, conditioned on line 10.

You can make it more complex from there.
You may want to think about conditioned such as checking if it's armed (equals, armed, 1).

**Reply by F1AWings** [reply]
Many thanks, I will try to learn the process
